web/js/MapUI/Components/CreateMap/CreateMap.js

```diff
@@ -0,0 +1,278 @@
+   */
+  const mapCreation = (proj, uiCopy) => {
+    const mapContainerEl = document.getElementById('wv-map');
+    const mapEl = document.createElement('div');
+    const id = `wv-map-${proj.id}`;
+
+    mapEl.setAttribute('id', id);
+    mapEl.setAttribute('data-proj', proj.id);
+    mapEl.classList.add('wv-map');
+    mapEl.style.display = 'none';
+    mapContainerEl.insertAdjacentElement('afterbegin', mapEl);
+
+    const scaleMetric = new OlControlScaleLine({
+      className: 'wv-map-scale-metric',
+      units: 'metric',
+    });
+    const scaleImperial = new OlControlScaleLine({
+      className: 'wv-map-scale-imperial',
+      units: 'imperial',
+    });
+    const rotateInteraction = new OlInteractionDragRotate({
+      condition: altKeyOnly,
+      duration: animationDuration,
+    });
+    const mobileRotation = new OlInteractionPinchRotate({
+      duration: animationDuration,
+    });
+    const map = new OlMap({
+      view: new OlView({
+        maxResolution: proj.resolutions[0],
+        projection: olProj.get(proj.crs),
+        center: proj.startCenter,
+        zoom: proj.startZoom,
+        maxZoom: proj.numZoomLevels,
+        enableRotation: true,
+        extent: proj.id === 'geographic' ? [-250, -90, 250, 90] : proj.maxExtent,
+        constrainOnlyCenter: true,
+      }),
+      target: id,
+      renderer: ['canvas'],
+      logo: false,
+      controls: [scaleMetric, scaleImperial],
+      interactions: [
+        doubleClickZoom,
+        new OlInteractionDragPan({
+          kinetic: new OlKinetic(-0.005, 0.05, 100),
+        }),
+        new OlInteractionPinchZoom({
+          duration: animationDuration,
+        }),
+        new OlInteractionMouseWheelZoom({
+          duration: animationDuration,
+        }),
+        new OlInteractionDragZoom({
+          duration: animationDuration,
+        }),
+      ],
+      loadTilesWhileAnimating: true,
+      loadTilesWhileInteracting: true,
+      maxTilesLoading: 32,
+    });
+    map.wv = {
+      scaleMetric,
+      scaleImperial,
+    };
+    map.proj = proj.id;
+
+    map.getView().on('change:resolution', () => {
+      events.trigger(MAP_MOVE_START);
+    });
+
+    // This component is inside the map viewport container. Allowing
+    // mouse move events to bubble up displays map coordinates--let those
+    // be blank when over a component.
+    document.querySelector('.wv-map-scale-metric').addEventListener('mousemove', (e) => e.stopPropagation());
+    document.querySelector('.wv-map-scale-imperial').addEventListener('mousemove', (e) => e.stopPropagation());
+
+    // Allow rotation by dragging for polar projections
+    if (proj.id !== 'geographic' && proj.id !== 'webmerc') {
+      map.addInteraction(rotateInteraction);
+      map.addInteraction(mobileRotation);
+    }
+
+    const onRotate = () => {
+      const radians = map.getView().getRotation();
+      updateRotation(radians);
+      const currentDeg = radians * (180.0 / Math.PI);
```

Comment:
We could eliminate the 180/PI in place of a constant to save doing this calculation repeatedly. If you go that route I would include a comment indicating how the value was derived.

Reply:
I think that they opted to originally write it like this because you can easily see how it is being calculated in the code. This function is only called when a user rotates the map in one of the polar views so it is not an expensive calculation. I would be open to talking about this one if you still think it should be changed to a variable.

Reply:
I googled this one & the suggestion was to declare `const PI_OVER_180 = Math.PI / 180;` & use that if it's being used frequently. 

https://stackoverflow.com/questions/20808698/how-to-calculate-math-pi-180-once-and-store-in-a-variable

You may be right about this having a negligible impact but a simple fix that provides (even minimal) improvement is worth doing IMO.

Reply:
I've updated with these changes since it still seems readable and easy enough to understand.